canonical-web-and-design/maas-ui
# refactor: update zone state shape in Redux

## Done
- This branch updates the state shape for the zone model in order to better handle actions and errors. It will serve as the blueprint for updating the rest of the models to the same shape.
- Supersedes https://github.com/canonical-web-and-design/maas-ui/pull/3503

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open Redux DevTools
- For the following actions, check that they work and that you like how the changes in state occur:
  - Go to the zone list and check that zones are fetched
  - Check that creating a zone works
  - Try to create another zone with the same name and check that an error shows
  - Go to a zone details page
  - Check that updating the zone works
  - Try updating the zone name to something that already exists and check that an error shows
  - Check that you can delete a zone and get redirected to the list

## Fixes

Fixes #3055
